channel.go

```diff
@@ -1,22 +1,14 @@
 package grpchan
 
 import (
-	"golang.org/x/net/context"
 	"google.golang.org/grpc"
 )
 
 // Channel is an abstraction of a GRPC transport. With corresponding generated
 // code, it can provide an alternate transport to the standard HTTP/2-based one.
 // For example, a Channel implementation could instead provide an HTTP 1.1-based
 // transport, or an in-process transport.
-type Channel interface {
-	// InvokeUnary executes a unary RPC, sending the given req message and populating
-	// the given resp with the server's reply.
-	Invoke(ctx context.Context, methodName string, req, resp interface{}, opts ...grpc.CallOption) error
-
-	// InvokeStream executes a streaming RPC.
-	NewStream(ctx context.Context, desc *grpc.StreamDesc, methodName string, opts ...grpc.CallOption) (grpc.ClientStream, error)
-}
+type Channel = grpc.ClientConnInterface
```

Comment:
Maybe this should be a separate PR?

But now that gRPC has its own version of this interface, using a type alias may make it easier to use grpchan in other contexts...

Reply:
SGTM assuming they're identical; otherwise it could break implementors.
Should we deprecate this symbol however, in favor fo the grpc type?

Reply:
Good call. I've pulled it out for now and will put it another branch, along with deprecation.